oceannavigator/frontend/src/components/OceanNavigator.jsx

```diff
@@ -209,21 +210,56 @@ class OceanNavigator extends React.Component {
 
       switch (key) {
         case "scale":
-          this.updateScale(key, value);
+          const newScaleState = {
+            variable_scale: value,
+          };
+
+          if (this.state.syncRanges) {
+            newScaleState["dataset_1"] = {
+              ...this.state.dataset_1,
+              variable_scale: value,
+            };
+          }
+
+          this.setState(newScaleState);
           return;
         case "scale_1":
+          const newScale_1State = {
+            dataset_1: {
+              ...this.state.dataset_1,
+              variable_scale: value,
+            }
+          };
 
           if (this.state.syncRanges) {
-            newState.scale = value;
-            newState.scale_1 = value;
+            newScale_1State["variable_scale"] = value;
           }
-          break;
-        case "dataset_0":
 
+          this.setState(newScale_1State);
+          return;
+        case "dataset_0":
           if (value.dataset !== this.state.dataset) {
             this.changeDataset(value.dataset, value);
             return;
           }
+          
+          // If variable is being changed...
+          if (value.variable !== this.state.variable) {
+            const variableFromCache = this.state.datasetVariablesCache[value.dataset]
+            .find(v => v.id === value.variable);
+
+            // Update interpolation
+            newState["dataset_0"]["options"] = {
+              ...this.state.options,
+              interpType: variableFromCache.interp?.interpType || this.DEF_INTERP_TYPE,
+              interpRadius: variableFromCache.interp?.interpRadius || this.DEF_INTERP_RADIUS_KM,
+              interpNeighbours: variableFromCache.interp?.interpNeighbours || this.DEF_INTERP_NUM_NEIGHBOURS,
```

Comment:
`?.`  https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining

Reply:
This syntax I'm using here is short-circuit assignment: `expression || expression` will use the first `expression` that returns a truthy value:

`null || this.DEF_INTERP_TYPE` will return `this.DEF_INTERP_TYPE`.

`"gaussian" || this.DEF_INTERP_TYPE` will return "gaussian".

https://dev.to/flexdinesh/short-circuit-assignment-in-javascript--4k80

Reply:
we use this conditional because in the API, variables don't define interp options by default (i.e. `interp === null`) since they're all using the same options by default.